src/realm/obj.cpp

```diff
@@ -1532,13 +1536,17 @@ Obj& Obj::set<ObjLink>(ColKey col_key, ObjLink target_link, bool is_default)
 Obj Obj::create_and_set_linked_object(ColKey col_key, bool is_default)
 {
     update_if_needed();
+    // Outgoing links from asymmetric objects are disallowed.
+    REALM_ASSERT(!get_table()->is_asymmetric());
```

Comment:
just checking that these asserts are validated somewhere else. REALM_ASSERT's are only included in debug builds and certain SDKs - so these are noops. Also, we should REALM_ASSERT only for invariants since they crash the program rather than raising an error for the user.

Reply:
this should actually not be possible, it's just a double guard.